src/ingest/src/main/scala/osmesa/ingest/GenerateVT.scala

```diff
@@ -0,0 +1,164 @@
+package osmesa
+
+import com.amazonaws.services.s3.model.CannedAccessControlList._
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.raster.rasterize.polygon._
+import geotrellis.spark._
+import geotrellis.spark.io.s3._
+import geotrellis.spark.tiling._
+import geotrellis.vector._
+import geotrellis.vectortile._
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.rdd.RDD
+
+import scala.concurrent._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+object GenerateVT {
+
+  lazy val logger = Logger.getRootLogger()
+
+  type VTF[G <: Geometry] = Feature[G, Map[String, Value]]
+
+  def save(vectorTiles: RDD[(SpatialKey, VectorTile)], zoom: Int, bucket: String, prefix: String) = {
+    vectorTiles
+      .mapValues(_.toBytes)
+      .saveToS3({ sk: SpatialKey => s"s3://${bucket}/${prefix}/${zoom}/${sk.col}/${sk.row}.mvt" },
+                putObjectModifier = { o => o.withCannedAcl(PublicRead) })
+  }
+
+  def apply[G <: Geometry](features: RDD[VTF[G]], layout: LayoutDefinition, layerName: String): RDD[(SpatialKey, VectorTile)] = {
+    val keyedGeoms = features.flatMap{ feat =>
+      val g = feat.geom
+      val keys = layout.mapTransform.keysForGeometry(g)
+      keys.map{ k => (k, (k, feat)) }
+    }
+
+    type FeatureTup = (Seq[VTF[Point]], Seq[VTF[MultiPoint]], Seq[VTF[Line]], Seq[VTF[MultiLine]], Seq[VTF[Polygon]], Seq[VTF[MultiPolygon]])
+
+    def timedIntersect[G <: Geometry](geom: G, ex: Extent) = {
+      val future = Future { geom.intersection(ex) }
+      Try(Await.result(future, 500 milliseconds)) match {
+        case Success(res) => res
+        case Failure(_) =>
+          logger.warn(s"Could not intersect $geom with $ex")
```

Comment:
If we could also log the feature id, that would be awesome, since we could then track down weird OSM geometries and figure out how to deal with them.